proxy/src/control/destination/background.rs

```diff
@@ -64,9 +93,9 @@ struct Background<T: HttpService<ResponseBody = RecvBody>> {
 }
 
 /// Holds the state of a single resolution.
-struct DestinationSet<T: HttpService<ResponseBody = RecvBody>> {
+struct DestinationSet<T: GetDst> {
     addrs: Exists<Cache<SocketAddr, Metadata>>,
-    query: Option<DestinationServiceQuery<T>>,
+    query: Option<Remote<PbUpdate, T::Error, T::Future, T::Stream>>,
```

Comment:
It's somewhat unfortunate that the reasonably clear `DestinationServiceQuery<T>` was replaced with this somewhat less clear type...maybe consider adding a 
```rust
type Query = Remote<PbUpdate, Self::Error, Self::Future, Self::Stream>;
``` 
type alias to `GetDst`, and then this can just be `Option<T::Query>`?